code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/integration/ldap/LDAPIdentityProvider.java

```diff
@@ -239,14 +239,20 @@ protected boolean isType(Attributes search, String type) {
         }
     }
 
-    protected NamingEnumeration<SearchResult> searchLdap(String query) {
+    protected NamingEnumeration<SearchResult> searchLdap(String query, String scope) {
         SearchControls controls = new SearchControls();
         LdapContext context = null;
         controls.setSearchScope(SUBTREE_SCOPE);
         NamingEnumeration<SearchResult> results;
         try {
             context = getServiceContext();
             results = context.search(getConstantsConfig().getDomain(), query, controls);
+            if (scope == getConstantsConfig().getGroupScope()) {
+                String groupSearchDomain = getConstantsConfig().getGroupSearchDomain();
+                if (StringUtils.isNotEmpty(groupSearchDomain) && StringUtils.isNotBlank(groupSearchDomain)) {
+                    results = context.search(getConstantsConfig().getGroupSearchDomain(), query, controls);
+                }
+            }
```

Comment:
We are doing an extra ldapsearch here. We should avoid the search over default domain when groupSearchDomain is specified [results = context.search(getConstantsConfig().getDomain(), query, controls); is not needed since you are overwriting results in second search]